proto/greptime/v1/database.proto

```diff
@@ -22,6 +22,8 @@ message RequestHeader {
   string schema = 2;
   // The `authorization` header, much like http's authorization header.
   AuthHeader authorization = 3;
+  // The `dbname` for the request
+  string dbname = 4;
```

Comment:
Since "dbname = catalog + '-' + schema", I suggest using "oneof":

```protobuf
message RequestHeader {
  message CatalogAndSchema {
    string catalog = 1;
    string schema = 2;
  }
  oneof database_ident {
    CatalogAndSchema catalog_and_schema = 1;
    string dbname = 2;
  }
  AuthHeader authorization = 3;
}
```

Reply:
This is preferred solution, to have consistency in proto semantic and implementation. However, I'm afraid this will become backward incompatible so I chose to append `dbname` to the header.

Reply:
Well, currently we have only one outside use of grpc client, and the one is in early adoption stage. I think we can afford to make this breaking change?

Reply:
Why don't we consider keeping the proto unchanged and just splitting/combining the `catalog-schema` at the upper level?

Reply:
Since the proto is also used in our intra-cluster communication, changing this can result in a incompatible upgrade from existing cluster..

Reply:
Do we really need the `dbname` in proto file? Can it be handled at the code level?

Reply:
Yes, but here I try to avoid leaking dbname generation pattern to client, for better extensibility. The pattern `catalog-schema` we are using for now might not be a perfect solution.